### PR TITLE
Assorted improvements

### PR DESCRIPTION
**Pull request by dszoboszlay**
## Summary

* List processes with their registered name
* Fix erlang:send/3 return value
* Add support for erlang:process_info/2 with a list argument

**Important**: the new test for `process_info/2` is currently broken. I get this output when running it, and I don't know how to fix it:

```
* The following pair of instructions is not explicitly marked as non-racing in Concuerror's internals:
  1) [{registered_name,[]},{group_leader,<0.74.0>}] = erlang:process_info(<0.85.0>, [registered_name,group_leader])
  2) true = erlang:register(foo, <0.85.0>)
 Please notify the developers to add info about this pair.
 You can run without '--assume_racing false' to treat them as racing.
 [{concuerror_dependencies,dependent_process_info,
      [{builtin_event,<0.82.0>,undefined,false,
           {erlang,process_info,[<0.85.0>,[registered_name,group_leader]]},
           [{registered_name,[]},{group_leader,<0.74.0>}],
           ok,false},
       {builtin_event,<0.85.0>,undefined,false,
           {erlang,register,[foo,<0.85.0>]},
           true,ok,false}],
      [{file,"src/concuerror_dependencies.erl"},{line,338}]},
  {concuerror_dependencies,'-dependent/3-lc$^3/1-3-',5,
      [{file,"src/concuerror_dependencies.erl"},{line,42}]},
  {concuerror_dependencies,dependent,3,
      [{file,"src/concuerror_dependencies.erl"},{line,42}]},
  {concuerror_scheduler,update_clock,4,
      [{file,"src/concuerror_scheduler.erl"},{line,887}]},
  {concuerror_scheduler,update_clock,5,
      [{file,"src/concuerror_scheduler.erl"},{line,870}]},
  {concuerror_scheduler,assign_happens_before,4,
      [{file,"src/concuerror_scheduler.erl"},{line,814}]},
  {concuerror_scheduler,plan_more_interleavings,1,
      [{file,"src/concuerror_scheduler.erl"},{line,774}]},
  {concuerror_scheduler,explore,1,
      [{file,"src/concuerror_scheduler.erl"},{line,238}]}]
```

I didn't want to add a broken result file, so I used a fake one that should work once this bug is fixed.

## Checklist

* [x] Has tests (or doesn't need them)
* [x] Updates CHANGELOG (or too minor)
* [x] References related Issues

### CODE REVIEW

**Comment by aronisstav** [comment]
Thanks for the contribution @dszoboszlay !

The failure in the `tests` suite is caused exactly because the suite is run with `--assume_racing false` which causes the exploration to stop if Concuerror doesn't have enough dependency information.

I will continue reviewing and updating this PR.

File: src/concuerror_callback.erl [hunk 1]
```diff
@@ -199,7 +199,13 @@ instrumented_call(Module, Name, Arity, Args, _Location,
...
       try
         Symbol = ets:lookup_element(Processes, Term, ?process_symbolic),
-        {{didit, Symbol}, Info}
+        PName = ets:lookup_element(Processes, Term, ?process_name),
+        Pretty = case PName =:= ?process_name_none of
+                   true -> Symbol;
+                   false -> lists:flatten(
+                              io_lib:format("<~s/~s>", [Symbol, PName]))
+                 end,
+        {{didit, Pretty}, Info}
```
Comment (aronisstav) [inline src/concuerror_callback.erl]: This works for tests where processes are 'deadlocked', bit won't work in general, as the logger accesses this information after an interleaving has been completely explored, at which point some processes will have unregistered their names.

**Comment by aronisstav** [comment]
Related: aronisstav/Concuerror#22

**Comment by dszoboszlay** [comment]
Rebased to fix merge conflict

**Comment by aronisstav** [comment]
Currently working on merging this. 😄

**Pull request merged**